src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -908,6 +908,26 @@ public static unsafe object IsInstanceOf(MethodTable* pTargetType, object obj)
                 return IsInstanceOfClass(pTargetType, obj);
         }
 
+        [RuntimeExport("RhTypeCast_IsSimpleInstanceOf")]
+        public static unsafe bool IsSimpleInstanceOf(MethodTable* typeHnd, object obj)
```

Comment:
Could you please change `ShouldTypedClauseCatchThisException` to call this method? It will ensure that both generic and non-generic casting uses the same logic.